cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -117,12 +117,16 @@ private String ensureIndexed(Index index) {
         try {
             if(index.indexName == null){
                 // generate a name for the index.
-                String indexName = this.generateIndexName();
+                String indexName = this.generateIndexName(listIndexesInDatabaseQueue());
```

Comment:
Pass in the `keySet()` here to be specific with the method's required input -- all `generateIndexName()` needs is the names of the indexes.

Also, for calling static methods, use `IndexCreator.generateIndexName` to make it clear. At least in IDEA, this is a warning too.